StardewArchipelago/GameModifications/CodeInjections/Bundles/BundleMenuInjection.cs

```diff
@@ -0,0 +1,194 @@
+                {
+                    myID = j + 1000,
+                    item = item,
+                    upNeighborID = -99998,
+                    rightNeighborID = -99998,
+                    leftNeighborID = -99998,
+                    downNeighborID = -99998,
+                });
+            }
+            var updateIngredientSlots = typeof(JunimoNoteMenu).GetMethod("updateIngredientSlots", BindingFlags.Instance | BindingFlags.NonPublic);
+            updateIngredientSlots.Invoke(__instance, System.Array.Empty<object>());
+            if (!Game1.options.SnappyMenus)
+            {
+                return false;
+            }
+            __instance.populateClickableComponentList();
+            if (__instance.inventory?.inventory != null)
+            {
+                for (int i = 0; i < __instance.inventory.inventory.Count; i++)
+                {
+                    if (__instance.inventory.inventory[i] != null)
+                    {
+                        if (__instance.inventory.inventory[i].downNeighborID == 101)
+                        {
+                            __instance.inventory.inventory[i].downNeighborID = -1;
+                        }
+                        if (__instance.inventory.inventory[i].leftNeighborID == -1)
+                        {
+                            __instance.inventory.inventory[i].leftNeighborID = 103;
+                        }
+                        if (__instance.inventory.inventory[i].upNeighborID >= 1000)
+                        {
+                            __instance.inventory.inventory[i].upNeighborID = 103;
+                        }
+                    }
+                }
+            }
+            __instance.currentlySnappedComponent = __instance.getComponentWithID(0);
+            __instance.snapCursorToCurrentSnappedComponent();
+            return false;
+        }
+        
+        //Transpiler way
+        public static IEnumerable<CodeInstruction> SkipObjectCheck(IEnumerable<CodeInstruction> instructions)
+        {
+            // We want to skip `if (dataOrErrorItem.HasTypeObject())`, just after `ParsedItemData dataOrErrorItem = ItemRegistry.GetDataOrErrorItem(representativeItemId);`
+            // In 1.6.8.24119, it looks like this
+            // IL_019c: ldloc.s 7
+            // IL_019e: call bool StardewValley.Extensions.ItemExtensions::HasTypeObject(class StardewValley.ItemTypeDefinitions.IHaveItemTypeId)
+            // IL_01a3: brfalse IL_0330
+
+            var getDataOrErrorItemMethodInfo = AccessTools.Method(typeof(ItemExtensions), nameof(ItemExtensions.HasTypeObject));
+
+            var foundMethod = -2; // -2 = not found, -1 = found, 0 = brfalse line, 1 = call line
+
+            Label? label = null;
+            var enumerator = instructions.GetEnumerator();
+            enumerator.MoveNext();
+            var previousInstruction = enumerator.Current;
+            while (enumerator.MoveNext())
```

Comment:
What is the point of manually using the enumerator instead of a foreach?

Reply:
Because I want to get the first instruction as previous_instruction and avoids having it null ever

Reply:
I feel like it would be way cleaner to have a null and just `continue;` the first iteration when it's null, but eh, I'm not attached to that. Good enough for me